packages/ui/src/theme/duplicationUtils.ts

```diff
@@ -0,0 +1,67 @@
+import { WebDesignToken } from './tokens/types/designToken';
+
+/**
+ * All of this code is to be removed in the next major version release and is being used to temporarily duplicate misnamed state tokens
+ * i.e. `hover` instead of `_hover`.  These tokens both map to the same css variable name which means we need special logic to remove duplicates
+ * should multiple instances of this variable exist.
+ */
+
+type TokenName = string;
+type TokenPath = string;
+interface DuplicateStateTokens {
+  [key: TokenName]: TokenPath;
+}
+
+/**
+ * This function takes in an array of WebDesignTokens and filters out the duplicate WebDesignTokens
+ * that are defined in the DUPLICATE_STATE_TOKENS list. This is designed to remove the duplicate tokens
+ * that were created due to inconsistent naming.  This should be removed when the tokens are removed in the
+ * next major version release.
+ * @param tokens
+ * @returns WebDesignTokens[]
+ */
+export function removeDuplicateStateTokens(
+  tokens: WebDesignToken[]
+): WebDesignToken[] {
+  const duplicateTokens = {};
+
+  return tokens
+    .reduce(
+      (prev: WebDesignToken[], current: WebDesignToken): WebDesignToken[] => {
+        if (DUPLICATE_STATE_TOKENS[current.name]) {
+          if (duplicateTokens[current.name]) {
+            duplicateTokens[current.name] =
+              current.path.join('.') === DUPLICATE_STATE_TOKENS[current.name]
+                ? current
+                : duplicateTokens[current.name];
+          } else {
+            duplicateTokens[current.name] = current;
+          }
+        } else {
+          prev.push(current);
+        }
+        return prev;
+      },
+      []
+    )
+    .concat(Object.values(duplicateTokens));
```

Comment:
This reads a bit more complicated than it might need to be, is there anything we can do to simplify? Can discuss offline if needed